grib/src/main/java/ucar/nc2/grib/grib2/table/Grib2Tables.java

```diff
@@ -460,9 +460,8 @@ public String getLevelNameShort(int id) {
 
   private TimeUnitConverter timeUnitConverter; // LOOK not really immutable
 
+  // Do not modify timeUnitConverter once it's been set
   public void setTimeUnitConverter(TimeUnitConverter timeUnitConverter) {
-    if (this.timeUnitConverter != null)
```

Comment:
Is it ok to remove this?

Reply:
It's a good question, I am not completely sure this is the right fix. The Grib1 version does not have this exception. And this function is getting called in the `Grib2CollectionBuilder::makeGroups`, so if you have a grib collection that you tell to recreate the index then that operation will fail due to this exception. It could be that there is a better fix for that case though.

Reply:
yeah I'm not sure what the reason for immutability is here other than just a general migration toward immutable objects...

Reply:
I will have one more look if I can see a different way to fix this

Reply:
I reverted this change and moved the `timeUnitConverter` to the `Grib2CollectionBuilder`, and deprecated the one in the `Grib2Tables`. I think this is a much better solution. The `Grib2Table` still has various `getForecast...` methods that now no longer do time unit conversions, but actually the callers of these methods handle the time units already.